src/Compilers/Core/Portable/CommandLine/CommandLineParser.cs

```diff
@@ -653,28 +648,29 @@ void parseResponseFile(string fullPath)
                     return;
                 }
 
-                for (var i = splitList.Count - 1; i >= 0; i--)
+                List<string>? notProcessedSplitList = null;
+                foreach (var newArg in splitList)
                 {
-                    var newArg = splitList[i];
                     // Ignores /noconfig option specified in a response file
-                    if (!string.Equals(newArg, "/noconfig", StringComparison.OrdinalIgnoreCase) && !string.Equals(newArg, "-noconfig", StringComparison.OrdinalIgnoreCase))
+                    if (string.Equals(newArg, "/noconfig", StringComparison.OrdinalIgnoreCase) || string.Equals(newArg, "-noconfig", StringComparison.OrdinalIgnoreCase))
```

Comment:
>if (string.Equals(newArg, "/noconfig", StringComparison.OrdinalIgnoreCase) || string.Equals(newArg, "-noconfig", StringComparison.OrdinalIgnoreCase)) [](http://example.com/codeflow?start=20&length=149)

Consider keeping the original condition to reduce the unnecessary changes and keep the diff to a necessary minimum.

Reply:
done